controllers/fio/controller.go

```diff
@@ -60,7 +66,22 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		return ctrl.Result{}, err
 	}
 
-	job := NewJob(&cr, configMap)
+	pvcName := cr.Spec.PersistentVolumeClaimName
+	if pvcName == nil {
+		pvc, err := NewPersistentVolumeClaim(&cr)
+		if err != nil {
+			r.K8S.EventRecorder.Eventf(crReference, corev1.EventTypeNormal, k8s.CreateSucceeded,
+				"Failed to create PVC. Error: ", err)
+		}
+		if pvc != nil {
```

Comment:
we could spare this if statement if we would add a check to line 70 for nil-ness of `cr.Spec.PersistentVolumeClaim`

Reply:
No, if `cr.Spec.PersistentVolumeClaim` is not nil, then we shouldn't create a new PVC in line 71, so this check can't go to line 70.